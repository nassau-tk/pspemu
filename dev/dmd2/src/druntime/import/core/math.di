// D import file generated from 'src\core\math.d'
module core.math;
version (LDC)
{
    public import ldc.intrinsics;

}
public 
{
    pure nothrow @safe real cos(real x);
    pure nothrow @safe real sin(real x);
    pure nothrow @safe long rndtol(real x);
    extern (C) real rndtonl(real x);

    nothrow pure @safe 
{
    float sqrt(float x);
    double sqrt(double x);
    real sqrt(real x);
}
    pure nothrow @safe real ldexp(real n, int exp);
        pure nothrow @safe real fabs(real x);
    pure nothrow @safe real rint(real x);
    pure nothrow @safe real yl2x(real x, real y);
    pure nothrow @safe real yl2xp1(real x, real y);
    }
