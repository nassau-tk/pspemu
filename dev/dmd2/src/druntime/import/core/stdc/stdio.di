// D import file generated from 'src\core\stdc\stdio.d'
module core.stdc.stdio;
private 
{
    import core.stdc.config;
    import core.stdc.stddef;
    import core.stdc.stdarg;
    version (FreeBSD)
{
    import core.sys.posix.sys.types;
}
}
extern (C) nothrow 
{
    version (Windows)
{
    enum 
{
BUFSIZ = 16384,
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 256,
TMP_MAX = 32767,
SYS_OPEN = 20,
}
    enum int _NFILE = 60;
    enum string _P_tmpdir = "\\";
    enum wstring _wP_tmpdir = "\\";
    enum int L_tmpnam = _P_tmpdir.length + 12;
}
else
{
    version (linux)
{
    enum 
{
BUFSIZ = 8192,
EOF = -1,
FOPEN_MAX = 16,
FILENAME_MAX = 4095,
TMP_MAX = 238328,
L_tmpnam = 20,
}
}
else
{
    version (OSX)
{
    enum 
{
BUFSIZ = 1024,
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 1024,
TMP_MAX = 308915776,
L_tmpnam = 1024,
}
    private 
{
    struct __sbuf
{
    ubyte* _base;
    int _size;
}
    struct __sFILEX
{
}
}
}
else
{
    version (FreeBSD)
{
    enum 
{
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 1024,
TMP_MAX = 308915776,
L_tmpnam = 1024,
}
    struct __sbuf
{
    ubyte* _base;
    int _size;
}
    alias _iobuf __sFILE;
    union __mbstate_t
{
    char[128] _mbstate8;
    long _mbstateL;
}
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    enum 
{
SEEK_SET,
SEEK_CUR,
SEEK_END,
}
    struct _iobuf
{
    align (1)version (Windows)
{
    char* _ptr;
    int _cnt;
    char* _base;
    int _flag;
    int _file;
    int _charbuf;
    int _bufsiz;
    int __tmpnum;
}
else
{
    version (linux)
{
    int _flags;
    char* _read_ptr;
    char* _read_end;
    char* _read_base;
    char* _write_base;
    char* _write_ptr;
    char* _write_end;
    char* _buf_base;
    char* _buf_end;
    char* _save_base;
    char* _backup_base;
    char* _save_end;
    void* _markers;
    _iobuf* _chain;
    int _fileno;
    int _blksize;
    int _old_offset;
    ushort _cur_column;
    byte _vtable_offset;
    char[1] _shortbuf;
    void* _lock;
}
else
{
    version (OSX)
{
    ubyte* _p;
    int _r;
    int _w;
    short _flags;
    short _file;
    __sbuf _bf;
    int _lbfsize;
    int* function(void*) _close;
    int* function(void*, char*, int) _read;
    fpos_t* function(void*, fpos_t, int) _seek;
    int* function(void*, char*, int) _write;
    __sbuf _ub;
    __sFILEX* _extra;
    int _ur;
    ubyte[3] _ubuf;
    ubyte[1] _nbuf;
    __sbuf _lb;
    int _blksize;
    fpos_t _offset;
}
else
{
    version (FreeBSD)
{
    ubyte* _p;
    int _r;
    int _w;
    short _flags;
    short _file;
    __sbuf _bf;
    int _lbfsize;
    void* _cookie;
    int function(void*) _close;
    int function(void*, char*, int) _read;
    fpos_t function(void*, fpos_t, int) _seek;
    int function(void*, in char*, int) _write;
    __sbuf _ub;
    ubyte* _up;
    int _ur;
    ubyte[3] _ubuf;
    ubyte[1] _nbuf;
    __sbuf _lb;
    int _blksize;
    fpos_t _offset;
    pthread_mutex_t _fl_mutex;
    pthread_t _fl_owner;
    int _fl_count;
    int _orientation;
    __mbstate_t _mbstate;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}

}
    alias shared(_iobuf) FILE;
    enum 
{
_F_RDWR = 3,
_F_READ = 1,
_F_WRIT = 2,
_F_BUF = 4,
_F_LBUF = 8,
_F_ERR = 16,
_F_EOF = 32,
_F_BIN = 64,
_F_IN = 128,
_F_OUT = 256,
_F_TERM = 512,
}
    version (Windows)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 64,
_IONBF = 4,
_IOREAD = 1,
_IOWRT = 2,
_IOMYBUF = 8,
_IOEOF = 16,
_IOERR = 32,
_IOSTRG = 64,
_IORW = 128,
_IOTRAN = 256,
_IOAPP = 512,
}
    extern shared void function() _fcloseallp;

    private extern shared FILE[_NFILE] _iob;


    shared stdin = &_iob[0];
    shared stdout = &_iob[1];
    shared stderr = &_iob[2];
    shared stdaux = &_iob[3];
    shared stdprn = &_iob[4];
}
else
{
    version (linux)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    extern shared FILE* stdin;

    extern shared FILE* stdout;

    extern shared FILE* stderr;

}
else
{
    version (OSX)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    private extern shared FILE* __stdinp;


    private extern shared FILE* __stdoutp;


    private extern shared FILE* __stderrp;


    alias __stdinp stdin;
    alias __stdoutp stdout;
    alias __stderrp stderr;
}
else
{
    version (FreeBSD)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    private extern shared FILE* __stdinp;


    private extern shared FILE* __stdoutp;


    private extern shared FILE* __stderrp;


    alias __stdinp stdin;
    alias __stdoutp stdout;
    alias __stderrp stderr;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    alias int fpos_t;
    int remove(in char* filename);
    int rename(in char* from, in char* to);
    FILE* tmpfile();
    char* tmpnam(char* s);
    int fclose(FILE* stream);
    int fflush(FILE* stream);
    FILE* fopen(in char* filename, in char* mode);
    FILE* freopen(in char* filename, in char* mode, FILE* stream);
    void setbuf(FILE* stream, char* buf);
    int setvbuf(FILE* stream, char* buf, int mode, size_t size);
    int fprintf(FILE* stream, in char* format,...);
    int fscanf(FILE* stream, in char* format,...);
    int sprintf(char* s, in char* format,...);
    int sscanf(in char* s, in char* format,...);
    int vfprintf(FILE* stream, in char* format, va_list arg);
    int vfscanf(FILE* stream, in char* format, va_list arg);
    int vsprintf(char* s, in char* format, va_list arg);
    int vsscanf(in char* s, in char* format, va_list arg);
    int vprintf(in char* format, va_list arg);
    int vscanf(in char* format, va_list arg);
    int printf(in char* format,...);
    int scanf(in char* format,...);
    int fgetc(FILE* stream);
    int fputc(int c, FILE* stream);
    char* fgets(char* s, int n, FILE* stream);
    int fputs(in char* s, FILE* stream);
    char* gets(char* s);
    int puts(in char* s);
    extern (D) 
{
    int getchar()
{
return getc(stdin);
}
    int putchar(int c)
{
return putc(c,stdout);
}
    int getc(FILE* stream)
{
return fgetc(stream);
}
    int putc(int c, FILE* stream)
{
return fputc(c,stream);
}
}
    int ungetc(int c, FILE* stream);
    size_t fread(void* ptr, size_t size, size_t nmemb, FILE* stream);
    size_t fwrite(in void* ptr, size_t size, size_t nmemb, FILE* stream);
    int fgetpos(FILE* stream, fpos_t* pos);
    int fsetpos(FILE* stream, in fpos_t* pos);
    int fseek(FILE* stream, c_long offset, int whence);
    c_long ftell(FILE* stream);
    version (Windows)
{
    extern (D) 
{
    void rewind(FILE* stream)
{
fseek(stream,0L,SEEK_SET);
stream._flag &= ~_IOERR;
}
    void clearerr(FILE* stream)
{
stream._flag &= ~(_IOERR | _IOEOF);
}
    int feof(FILE* stream)
{
return stream._flag & _IOEOF;
}
    int ferror(FILE* stream)
{
return stream._flag & _IOERR;
}
}
    int _snprintf(char* s, size_t n, in char* fmt,...);
    alias _snprintf snprintf;
    int _vsnprintf(char* s, size_t n, in char* format, va_list arg);
    alias _vsnprintf vsnprintf;
}
else
{
    version (linux)
{
    void rewind(FILE* stream);
    void clearerr(FILE* stream);
    int feof(FILE* stream);
    int ferror(FILE* stream);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    version (OSX)
{
    void rewind(FILE*);
    void clearerr(FILE*);
    int feof(FILE*);
    int ferror(FILE*);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    version (FreeBSD)
{
    void rewind(FILE*);
    void clearerr(FILE*);
    int feof(FILE*);
    int ferror(FILE*);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    void perror(in char* s);
}

