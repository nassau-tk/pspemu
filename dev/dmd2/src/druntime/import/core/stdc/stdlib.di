// D import file generated from 'src\core\stdc\stdlib.d'
module core.stdc.stdlib;
private import core.stdc.config;

public import core.stdc.stddef;

extern (C) nothrow 
{
    struct div_t
{
    int quot;
    int rem;
}
    struct ldiv_t
{
    int quot;
    int rem;
}
    struct lldiv_t
{
    long quot;
    long rem;
}
    enum EXIT_SUCCESS = 0;
    enum EXIT_FAILURE = 1;
    enum MB_CUR_MAX = 1;
    version (Windows)
{
    enum RAND_MAX = 32767;
}
else
{
    version (linux)
{
    enum RAND_MAX = 2147483647;
}
else
{
    version (OSX)
{
    enum RAND_MAX = 2147483647;
}
else
{
    version (FreeBSD)
{
    enum RAND_MAX = 2147483647;
}
else
{
    version (Solaris)
{
    enum RAND_MAX = 32767;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
}
    double atof(in char* nptr);
    int atoi(in char* nptr);
    c_long atol(in char* nptr);
    long atoll(in char* nptr);
    double strtod(in char* nptr, char** endptr);
    float strtof(in char* nptr, char** endptr);
    real strtold(in char* nptr, char** endptr);
    c_long strtol(in char* nptr, char** endptr, int base);
    long strtoll(in char* nptr, char** endptr, int base);
    c_ulong strtoul(in char* nptr, char** endptr, int base);
    ulong strtoull(in char* nptr, char** endptr, int base);
    int rand();
    void srand(uint seed);
    void* malloc(size_t size);
    void* calloc(size_t nmemb, size_t size);
    void* realloc(void* ptr, size_t size);
    void free(void* ptr);
    void abort();
    void exit(int status);
    int atexit(void function() func);
    void _Exit(int status);
    char* getenv(in char* name);
    int system(in char* string);
    void* bsearch(in void* key, in void* base, size_t nmemb, size_t size, int function(in void*, in void*) compar);
    void qsort(void* base, size_t nmemb, size_t size, int function(in void*, in void*) compar);
    pure int abs(int j);

    pure c_long labs(c_long j);

    pure long llabs(long j);

    div_t div(int numer, int denom);
    ldiv_t ldiv(c_long numer, c_long denom);
    lldiv_t lldiv(long numer, long denom);
    int mblen(in char* s, size_t n);
    int mbtowc(wchar_t* pwc, in char* s, size_t n);
    int wctomb(char* s, wchar_t wc);
    size_t mbstowcs(wchar_t* pwcs, in char* s, size_t n);
    size_t wcstombs(char* s, in wchar_t* pwcs, size_t n);
    version (DigitalMars)
{
    void* alloca(size_t size);
}
}

