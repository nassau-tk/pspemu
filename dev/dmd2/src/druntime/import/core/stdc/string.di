// D import file generated from 'src\core\stdc\string.d'
module core.stdc.string;
private import core.stdc.stddef;

extern (C) nothrow @system 
{
    pure void* memchr(in void* s, int c, size_t n);

    pure int memcmp(in void* s1, in void* s2, size_t n);

    void* memcpy(void* s1, in void* s2, size_t n);
    void* memmove(void* s1, in void* s2, size_t n);
    void* memset(void* s, int c, size_t n);
    char* strcpy(char* s1, in char* s2);
    char* strncpy(char* s1, in char* s2, size_t n);
    char* strcat(char* s1, in char* s2);
    char* strncat(char* s1, in char* s2, size_t n);
    pure int strcmp(in char* s1, in char* s2);

    int strcoll(in char* s1, in char* s2);
    pure int strncmp(in char* s1, in char* s2, size_t n);

    size_t strxfrm(char* s1, in char* s2, size_t n);
    pure char* strchr(in char* s, int c);

    pure size_t strcspn(in char* s1, in char* s2);

    char* strpbrk(in char* s1, in char* s2);
    pure char* strrchr(in char* s, int c);

    pure size_t strspn(in char* s1, in char* s2);

    pure char* strstr(in char* s1, in char* s2);

    char* strtok(char* s1, in char* s2);
    char* strerror(int errnum);
    pure size_t strlen(in char* s);

}


