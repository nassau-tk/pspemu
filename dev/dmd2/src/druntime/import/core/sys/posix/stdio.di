// D import file generated from 'src\core\sys\posix\stdio.d'
module core.sys.posix.stdio;
private import core.sys.posix.config;

public import core.stdc.stdio;

public import core.sys.posix.sys.types;

extern (C) 
{
    version (linux)
{
    static if(__USE_LARGEFILE64)
{
    int fgetpos64(FILE*, fpos_t*);
    alias fgetpos64 fgetpos;
    FILE* fopen64(in char*, in char*);
    alias fopen64 fopen;
    FILE* freopen64(in char*, in char*, FILE*);
    alias freopen64 freopen;
    int fseek64(FILE*, c_long, int);
    alias fseek64 fseek;
    int fsetpos64(FILE*, in fpos_t*);
    alias fsetpos64 fsetpos;
    FILE* tmpfile64();
    alias tmpfile64 tmpfile;
}
else
{
    int fgetpos(FILE*, fpos_t*);
    FILE* fopen(in char*, in char*);
    FILE* freopen(in char*, in char*, FILE*);
    int fseek(FILE*, c_long, int);
    int fsetpos(FILE*, in fpos_t*);
    FILE* tmpfile();
}
}
    version (linux)
{
    enum L_ctermid = 9;
    static if(__USE_FILE_OFFSET64)
{
    int fseeko64(FILE*, off_t, int);
    alias fseeko64 fseeko;
}
else
{
    int fseeko(FILE*, off_t, int);
}
    static if(__USE_LARGEFILE64)
{
    off_t ftello64(FILE*);
    alias ftello64 ftello;
}
else
{
    off_t ftello(FILE*);
}
}
else
{
    version (Posix)
{
    int fseeko(FILE*, off_t, int);
    off_t ftello(FILE*);
}
}
    version (Posix)
{
    char* ctermid(char*);
    FILE* fdopen(int, in char*);
    int fileno(FILE*);
    char* gets(char*);
    int pclose(FILE*);
    FILE* popen(in char*, in char*);
}
    version (linux)
{
    void flockfile(FILE*);
    int ftrylockfile(FILE*);
    void funlockfile(FILE*);
    int getc_unlocked(FILE*);
    int getchar_unlocked();
    int putc_unlocked(int, FILE*);
    int putchar_unlocked(int);
}
    version (linux)
{
    enum P_tmpdir = "/tmp";
    char* tempnam(in char*, in char*);
}
}
