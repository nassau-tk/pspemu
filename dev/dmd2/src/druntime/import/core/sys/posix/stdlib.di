// D import file generated from 'src\core\sys\posix\stdlib.d'
module core.sys.posix.stdlib;
private import core.sys.posix.config;

public import core.stdc.stdlib;

public import core.sys.posix.sys.wait;

extern (C) 
{
    version (linux)
{
    int posix_memalign(void**, size_t, size_t);
}
else
{
    version (FreeBSD)
{
    int posix_memalign(void**, size_t, size_t);
}
}
    version (linux)
{
    int setenv(in char*, in char*, int);
    int unsetenv(in char*);
    void* valloc(size_t);
}
else
{
    version (OSX)
{
    int setenv(in char*, in char*, int);
    int unsetenv(in char*);
    void* valloc(size_t);
}
else
{
    version (FreeBSD)
{
    int setenv(in char*, in char*, int);
    int unsetenv(in char*);
    void* valloc(size_t);
}
}
}
    version (linux)
{
    int rand_r(uint*);
}
else
{
    version (OSX)
{
    int rand_r(uint*);
}
else
{
    version (FreeBSD)
{
    int rand_r(uint*);
}
}
}
    version (linux)
{
    c_long a64l(in char*);
    double drand48();
    char* ecvt(double, int, int*, int*);
    double erand48(ref ushort[3]);
    char* fcvt(double, int, int*, int*);
    char* gcvt(double, int, char*);
    int getsubopt(char**, in char**, char**);
    int grantpt(int);
    char* initstate(uint, char*, size_t);
    c_long jrand48(ref ushort[3]);
    char* l64a(c_long);
    void lcong48(ref ushort[7]);
    c_long lrand48();
    char* mktemp(char*);
    c_long mrand48();
    c_long nrand48(ref ushort[3]);
    int posix_openpt(int);
    char* ptsname(int);
    int putenv(char*);
    c_long random();
    char* realpath(in char*, char*);
    ushort seed48(ref ushort[3]);
    void setkey(in char*);
    char* setstate(in char*);
    void srand48(c_long);
    void srandom(uint);
    int unlockpt(int);
    static if(__USE_LARGEFILE64)
{
    int mkstemp64(char*);
    alias mkstemp64 mkstemp;
}
else
{
    int mkstemp(char*);
}
}
else
{
    version (OSX)
{
    c_long a64l(in char*);
    double drand48();
    char* ecvt(double, int, int*, int*);
    double erand48(ref ushort[3]);
    char* fcvt(double, int, int*, int*);
    char* gcvt(double, int, char*);
    int getsubopt(char**, in char**, char**);
    int grantpt(int);
    char* initstate(uint, char*, size_t);
    c_long jrand48(ref ushort[3]);
    char* l64a(c_long);
    void lcong48(ref ushort[7]);
    c_long lrand48();
    char* mktemp(char*);
    int mkstemp(char*);
    c_long mrand48();
    c_long nrand48(ref ushort[3]);
    int posix_openpt(int);
    char* ptsname(int);
    int putenv(char*);
    c_long random();
    char* realpath(in char*, char*);
    ushort seed48(ref ushort[3]);
    void setkey(in char*);
    char* setstate(in char*);
    void srand48(c_long);
    void srandom(uint);
    int unlockpt(int);
}
else
{
    version (FreeBSD)
{
    c_long a64l(in char*);
    double drand48();
    double erand48(ref ushort[3]);
    int getsubopt(char**, in char**, char**);
    int grantpt(int);
    char* initstate(uint, char*, size_t);
    c_long jrand48(ref ushort[3]);
    char* l64a(c_long);
    void lcong48(ref ushort[7]);
    c_long lrand48();
    char* mktemp(char*);
    int mkstemp(char*);
    c_long mrand48();
    c_long nrand48(ref ushort[3]);
    int posix_openpt(int);
    char* ptsname(int);
    int putenv(char*);
    c_long random();
    char* realpath(in char*, char*);
    ushort seed48(ref ushort[3]);
    void setkey(in char*);
    char* setstate(in char*);
    void srand48(c_long);
    void srandom(uint);
    int unlockpt(int);
}
}
}
}
