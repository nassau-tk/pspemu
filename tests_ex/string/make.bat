@ECHO OFF
CALL ..\prepare.bat
psp-gcc -I. -I"%PSPSDK%/psp/sdk/include" -L. -L"%PSPSDK%/psp/sdk/lib" -D_PSP_FW_VERSION=150 -Wall -g -O0 string.c ../emits.c -lpspdebug -lpspge -lpspdisplay -lpspsdk -lc -lpspuser -lpspkernel -o string.elf
IF EXIST string.elf (
	psp-fixup-imports string.elf
)