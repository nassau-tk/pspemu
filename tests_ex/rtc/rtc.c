//#pragma compile, "%PSPSDK%/bin/psp-gcc" -I. -I"%PSPSDK%/psp/sdk/include" -L. -L"%PSPSDK%/psp/sdk/lib" -D_PSP_FW_VERSION=150 -Wall -g rtc.c ../common/emits.c -lpspsdk -lc -lpspuser -lpspkernel -lpsprtc -o rtc.elf
//#pragma compile, "%PSPSDK%/bin/psp-fixup-imports" rtc.elf

#include <pspkernel.h>
#include <psprtc.h>

#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

//#include "../common/emits.h"

#define eprintf(...) pspDebugScreenPrintf(__VA_ARGS__); Kprintf(__VA_ARGS__);

PSP_MODULE_INFO("rtctest", 0, 1, 1);
PSP_MAIN_THREAD_ATTR(THREAD_ATTR_USER);

/**
 * Check that getCurrentTick works fine.
 *
 * @TODO: Currently sceKernelDelayThread only work with ms.
 *        It should check that work with microseconds precission.
 */
void checkGetCurrentTick() {
	u64 tick0, tick1;
	int microseconds = 2 * 1000; // 2ms

	eprintf("Checking sceRtcGetCurrentTick\n");

	sceRtcGetCurrentTick(&tick0);
	{
		sceKernelDelayThread(microseconds);
	}
	sceRtcGetCurrentTick(&tick1);
	
	eprintf("%d\n", (tick1 - tick0) >= microseconds);
}

void checkDaysInMonth() {
	eprintf("Checking sceRtcGetDaysInMonth\n");
	eprintf("sceRtcGetDaysInMonth:2010, 4\n");
	eprintf("%d\n", sceRtcGetDaysInMonth(2010, 4));
}

void checkDayOfWeek() {
	eprintf("Checking sceRtcGetDayOfWeek\n");
	eprintf("sceRtcGetDayOfWeek:2010, 4, 27\n");
	eprintf("%d\n", sceRtcGetDayOfWeek(2010, 4, 27));
}

int main() {
	pspDebugScreenInit();

	checkGetCurrentTick();
	checkDaysInMonth();
	checkDayOfWeek();
	
	return 0;
}
