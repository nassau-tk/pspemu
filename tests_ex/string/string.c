#include <pspkernel.h>
#include <pspthreadman.h>
#include <stdio.h>
#include <string.h>

PSP_MODULE_INFO("STRING TEST", 0, 1, 1);
PSP_MAIN_THREAD_ATTR(THREAD_ATTR_USER | THREAD_ATTR_VFPU);

float value = 13.0;

int main(int argc, char** argv) {
	char buffer[128];
	
	sprintf(buffer, "%d", (u32)0);
	Kprintf("%s\n", buffer);
	
	sprintf(buffer, "%d", (u32)100000);
	Kprintf("%s\n", buffer);

	sprintf(buffer, "%lld", (u64)9);
	Kprintf("%s\n", buffer);

	sprintf(buffer, "%lld", (u64)100000);
	Kprintf("%s\n", buffer);

	sprintf(buffer, "%.2f", value);
	Kprintf("%s\n", buffer);

	sceKernelExitGame();
	return 0;
}
