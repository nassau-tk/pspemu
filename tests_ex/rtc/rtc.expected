Checking sceRtcGetCurrentTick
1
Checking sceRtcGetDaysInMonth
sceRtcGetDaysInMonth:2010, 4
30
Checking sceRtcGetDayOfWeek
sceRtcGetDayOfWeek:2010, 4, 27
1
